backend/src/test/java/com/engineering/thesis/backend/controllerIntegrationTests/AppointmentEPTests.java

```diff
@@ -0,0 +1,191 @@
+
+import com.engineering.thesis.backend.controller.AppointmentController;
+import com.engineering.thesis.backend.controllerIntegrationTests.configration.MockConfiguration;
+import com.engineering.thesis.backend.model.Appointment;
+import com.engineering.thesis.backend.model.Doctor;
+import com.engineering.thesis.backend.model.Patient;
+import com.engineering.thesis.backend.model.User;
+import com.engineering.thesis.backend.serviceImpl.AppointmentServiceImpl;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.runner.RunWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static com.engineering.thesis.backend.controllerIntegrationTests.role.RoleProcessors.RulesMap;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@ExtendWith(MockitoExtension.class)
+@WebMvcTest(value = AppointmentController.class)
+class AppointmentEPTests extends MockConfiguration {
+
+    @MockBean
+    private AppointmentServiceImpl appointmentService;
+
+    @Test
+    public void selectAllShouldReturnAppointmentWhenProperRoleIsSelected() throws Exception {
+        final User userDoctor1 = new User(1l, "Tom", "Kowalsky", "dsadzx1xczsa@osom.com", "1I@wsdas", "ROLE_DOCTOR", true);
+        final User userDoctor2 = new User(2l, "Tom", "Kowalsky", "dsadzx2xczsa@osom.com", "1I@wsdas", "ROLE_DOCTOR", true);
+        final Doctor doctor1 = new Doctor(1l, userDoctor1, "Cardiology");
+        final Doctor doctor2 = new Doctor(2l, userDoctor2, "Cardiology");
+        final User userPatient1 = new User(4l, "Tom", "Kowalsky", "dsad12ssssa@osom.com", "1I@wsdas", "ROLE_PATIENT", true);
+        final User userPatient2 = new User(5l, "Tom", "Kowalsky", "dsad32ssssa@osom.com", "1I@wsdas", "ROLE_PATIENT", true);
+        final Patient patient1 = new Patient(1l, userPatient1, true);
+        final Patient patient2 = new Patient(2l, userPatient2, true);
+        when(appointmentService.selectAll())
+                .thenReturn(List.of(
+                        new Appointment(1l, patient1, doctor1, 1000L, LocalDateTime.now()),
+                        new Appointment(2l, patient2, doctor2, 2000, LocalDateTime.now())
+                ));
+        RulesMap().forEach((name, Role) -> {
```

Comment:
We check if our roles works good and handling proper response. This iteration will reduce code for now and add some flexability in future.